app/build.gradle

```diff
@@ -17,9 +18,18 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+
+    compileOptions {
+        // Flag to enable support for the new language APIs
+        coreLibraryDesugaringEnabled false
+        // Sets Java compatibility to Java 8
+        sourceCompatibility JavaVersion.VERSION_1_8
```

Comment:
We have avoided this in the past because if we upgrade to Java 8 we force people who consume this library to upgrade to Java 8 in their apps.  Statistics show that Java 8 support is actually not at all popular with Android developers, looks like almost everyone is on Java 7 or Kotlin.

Is there a way we can have Java 8 within this library but desugar the AAR we publish so that consumers only require Java 7?

Reply:
I think we are having Java8+ support on Android by D8/R8, not Java7+.
https://developer.android.com/studio/write/java8-support#library-desugaring

So, Java8+ feature will be only available when the project is at Java8.
And eventually, all projects will be at Java8 level. (incl Kotlin projects)

Reply:
But agree. Some existing user may be forced to upgrade to Java8 because of this change.

Reply:
Yeah right now we just don't want to force anyone to upgrade.  That said if you want to scope this change only to the `app` module I am happy to update the sample app to Java 8.

Reply:
Updated. App Sample at Java8 and Library at Java7.